### PR TITLE
Fix an assertion failure in resgroup.

### PR DESCRIPTION
**Pull request by pivotal-ning-yu**
Suppose a resgroup already has a session assigned but no query running.
In another session (not assigned to this resgroup) drop this resgroup
and recreate it, then an assertion failure would be triggered.

The cause is that the old resgroup's struct in shm is reused for the new
resgroup, so on a QE who only owns a weak ref to this struct may find the
struct's resgroup id different than its expection.

To fix it we add the checks for such cases.

### CODE REVIEW

**Comment by pivotal-ning-yu** [comment]
@danielgustafsson Thank you for your review ;)

File: src/test/isolation2/expected/resgroup_recreate.out [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+1: BEGIN;
+BEGIN
+1: END;
+END
+
+DROP ROLE r1;
+DROP
+DROP RESOURCE GROUP rg1;
+DROP
+
```
Comment (xiong-gang) [inline src/test/isolation2/expected/resgroup_recreate.out]: empty line

Comment (pivotal-ning-yu) [inline src/test/isolation2/expected/resgroup_recreate.out]: thanks, will remove it.

Comment (pivotal-ning-yu) [inline src/test/isolation2/expected/resgroup_recreate.out]: This empty line is removed in a new commit.

**Comment by pivotal-ning-yu** [comment]
Thanks for all the review, we have merged this PR just now as it's needed for some resgroup tests on pipeline.

**Pull request merged**